ckanext/protected_resources/controllers.py

```diff
@@ -0,0 +1,38 @@
+from ckan.lib.base import BaseController, c
+from ckan import model
+import ckan.plugins.toolkit as tk
+
+class ProtectedResourceController(BaseController):
+
+    def unlock(self, dataset_id, resource_id):
+        try:
+            context = {'model': model, 'user': c.user}
+            if not tk.check_access('protected_resource_lock', context, {'resource_id': resource_id}):
+                tk.abort(401, 'You are not authorized to lock resources')
```

Comment:
5/10
This message may be unclear as the user attempted to `unlock` the resource.
Prefer using `unlock` in messages in this method.

Reply:
will fix